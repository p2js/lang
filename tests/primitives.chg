fn t1(): u8 { return 255u8; }
fn t2(): u16 { return 65535u16; }
fn t3(): u32 { return 4294967296u32; }
fn t4(): u64 { return 18446744073709551615u64; }

fn t5(): i8 { return 127i8; }
fn t55(): i8 { return 5i8; }
fn t6(): i16 { return 32767i16; }
fn t7(): i32 { return 2147483647i32; }
fn t8(): i64 { return 9223372036854775807i64; }

fn int_inference(): i64 { return 5; }
fn int_inference(): i64 { return 5 * 5; }

fn t9(): bool { return true; }
fn t10(): bool { return false; }

fn main(): i32 {
    t1(); t2(); t3(); t4(); t5(); t55(); t6(); t7(); t8(); t9(); t10();
    return 1;
}
