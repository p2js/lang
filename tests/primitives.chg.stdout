expr @ 1:22 1:24 => u8
expr @ 1:15 1:24 => never
expr @ 1:13 2:0 => void
expr @ 2:23 2:28 => u16
expr @ 2:16 2:28 => never
expr @ 2:14 3:0 => void
expr @ 3:23 3:33 => u32
expr @ 3:16 3:33 => never
expr @ 3:14 4:0 => void
expr @ 4:23 4:43 => u64
expr @ 4:16 4:43 => never
expr @ 4:14 5:0 => void
expr @ 6:22 6:24 => i8
expr @ 6:15 6:24 => never
expr @ 6:13 7:0 => void
expr @ 7:23 7:28 => i16
expr @ 7:16 7:28 => never
expr @ 7:14 8:0 => void
expr @ 8:23 8:33 => i32
expr @ 8:16 8:33 => never
expr @ 8:14 9:0 => void
expr @ 9:23 9:42 => i64
expr @ 9:16 9:42 => never
expr @ 9:14 10:0 => void
expr @ 11:24 11:28 => bool
expr @ 11:17 11:28 => never
expr @ 11:15 12:0 => void
expr @ 12:25 12:30 => bool
expr @ 12:18 12:30 => never
expr @ 12:16 13:0 => void
expr @ 15:5 15:7 => fn t1(...)
expr @ 15:5 15:9 => u8
expr @ 15:11 15:13 => fn t2(...)
expr @ 15:11 15:15 => u16
expr @ 15:17 15:19 => fn t3(...)
expr @ 15:17 15:21 => u32
expr @ 15:23 15:25 => fn t4(...)
expr @ 15:23 15:27 => u64
expr @ 15:29 15:31 => fn t5(...)
expr @ 15:29 15:33 => i8
expr @ 15:35 15:37 => fn t6(...)
expr @ 15:35 15:39 => i16
expr @ 15:41 15:43 => fn t7(...)
expr @ 15:41 15:45 => i32
expr @ 15:47 15:49 => fn t8(...)
expr @ 15:47 15:51 => i64
expr @ 15:53 15:55 => fn t9(...)
expr @ 15:53 15:57 => bool
expr @ 15:59 15:62 => fn t10(...)
expr @ 15:59 15:64 => bool
expr @ 16:12 16:13 => i32
expr @ 16:5 16:13 => never
expr @ 14:16 18:1 => void



define i8 @t1() {
prologue:
%l.0 = alloca i8
br label %bb.0

bb.0:
store i8 25, i8* %l.0
%ret.0 = load i8, i8* %l.0
ret i8 %ret.0

}

define i16 @t2() {
prologue:
%l.0 = alloca i16
br label %bb.0

bb.0:
store i16 65535, i16* %l.0
%ret.0 = load i16, i16* %l.0
ret i16 %ret.0

}

define i32 @t3() {
prologue:
%l.0 = alloca i32
br label %bb.0

bb.0:
store i32 4294967296, i32* %l.0
%ret.0 = load i32, i32* %l.0
ret i32 %ret.0

}

define i64 @t4() {
prologue:
%l.0 = alloca i64
br label %bb.0

bb.0:
store i64 18446744073709552000, i64* %l.0
%ret.0 = load i64, i64* %l.0
ret i64 %ret.0

}

define i8 @t5() {
prologue:
%l.0 = alloca i8
br label %bb.0

bb.0:
store i8 12, i8* %l.0
%ret.0 = load i8, i8* %l.0
ret i8 %ret.0

}

define i16 @t6() {
prologue:
%l.0 = alloca i16
br label %bb.0

bb.0:
store i16 32767, i16* %l.0
%ret.0 = load i16, i16* %l.0
ret i16 %ret.0

}

define i32 @t7() {
prologue:
%l.0 = alloca i32
br label %bb.0

bb.0:
store i32 2147483647, i32* %l.0
%ret.0 = load i32, i32* %l.0
ret i32 %ret.0

}

define i64 @t8() {
prologue:
%l.0 = alloca i64
br label %bb.0

bb.0:
store i64 9223372036854776000, i64* %l.0
%ret.0 = load i64, i64* %l.0
ret i64 %ret.0

}

define i1 @t9() {
prologue:
%l.0 = alloca i1
br label %bb.0

bb.0:
store i1 true, i1* %l.0
%ret.0 = load i1, i1* %l.0
ret i1 %ret.0

}

define i1 @t10() {
prologue:
%l.0 = alloca i1
br label %bb.0

bb.0:
store i1 false, i1* %l.0
%ret.0 = load i1, i1* %l.0
ret i1 %ret.0

}

define i32 @main() {
prologue:
%l.0 = alloca i32
br label %bb.0

bb.0:
%l.1 = call i8 @t1()
br label %bb.1

bb.1:
%l.2 = call i16 @t2()
br label %bb.2

bb.2:
%l.3 = call i32 @t3()
br label %bb.3

bb.3:
%l.4 = call i64 @t4()
br label %bb.4

bb.4:
%l.5 = call i8 @t5()
br label %bb.5

bb.5:
%l.6 = call i16 @t6()
br label %bb.6

bb.6:
%l.7 = call i32 @t7()
br label %bb.7

bb.7:
%l.8 = call i64 @t8()
br label %bb.8

bb.8:
%l.9 = call i1 @t9()
br label %bb.9

bb.9:
%l.10 = call i1 @t10()
br label %bb.10

bb.10:
store i32 1, i32* %l.0
%ret.0 = load i32, i32* %l.0
ret i32 %ret.0

}


